### PR TITLE
ci(github): fix make dev/tools

### PR DESCRIPTION
**Pull request by lahabana**
Calling make dev/tools multiple times would fail

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/dependencies/go-deps.sh [hunk 1]
```diff
@@ -12,6 +12,7 @@ PGKUMADOC=github.com/kumahq/protoc-gen-kumadoc@$(go list -f '{{.Version}}' -m gi
 GINKGO=github.com/onsi/ginkgo/v2/ginkgo@$(go list -f '{{.Version}}' -m github.com/onsi/ginkgo/v2)
 CONTROLLER_GEN=sigs.k8s.io/controller-tools/cmd/controller-gen@$(go list -f '{{.Version}}' -m sigs.k8s.io/controller-tools)
 
+> mk/dependencies/go-deps.versions
```
Comment (michaelbeaumont) [inline mk/dependencies/go-deps.sh:15]: Uhhh what does this mean? :grinning:

Comment (jijiechen) [reply mk/dependencies/go-deps.sh:15]: This statement resets the file content to empty, identical to the following:
```
echo '' > mk/dependencies/go-deps.versions
```

**Comment by jijiechen** [comment]
Let me add some background to this PR:
This PR is a fix to a [previous PR](https://github.com/kumahq/kuma/pull/8631) that tries to improve the cache in the CI. 
The PR collects all versions of the go tools involved and save it to the file `mk/dependencies/go-deps.versions` and using its content as part of the cache key. 

However, there is a bug in it: when executed multiple times, the file will contain duplicated entries, and will potentially lead to  unexpected cache misses.
So the fix here is to reset the file content every time before the installation to mitgrate the duplication.

File: mk/dependencies/go-deps.sh (hunk 1 above)
Comment (jijiechen) [inline mk/dependencies/go-deps.sh:15]:
```suggestion
echo '' > mk/dependencies/go-deps.versions
```

Comment (jijiechen) [reply mk/dependencies/go-deps.sh:15]: This is to pass the shell check, so fixes
https://github.com/kumahq/kuma/actions/runs/7221117859/job/19675420378?pr=8635

**Pull request merged**